jax/experimental/jax2tf/README.md

```diff
@@ -1229,6 +1229,58 @@ DeviceArray data or for np.ndarray that are aligned on 16-byte
 boundaries) and on GPU (for DeviceArray).
 The zero-copy does not yet work on TPU.
 
+`call_tf` works even in presence of shape polymorphism, but in that case
+the user must pass the `output_shape_dtype` parameter to `call_tf` to declare
+the expected output shape. This allows JAX tracing to know the shape and
+dtype of the results so that it can continue tracing the rest of the program.
+When `output_shape_dtype` is not given (the default case), `call_tf` will
+form a `tf.Graph` for the called TF function and will use the inferred
+type and shape. However, in presence of dynamic shape the inferred TF
+type will contain `None` for the dynamic dimensions, which is not enough
+information for JAX shape polymorphism.
+
+For example:
+
+```python
+def fun_jax(x):
+  y_shape = (x.shape[0] * 2, y.shape[1:])
+  y = jax2tf.call_tf(
+      lambda x: tf.concat([x, x], axis=0),
+      output_shape_dype=jax.ShapeDtypeStruct(y_shape, x.dtype))(x)
```

Comment:
nit:  The better idea about the usage is that we write a tf_func output_infer_shape function since output_shape_and_dtype should depend on inputs.

Example
```python
def fun_jax(x):
  output_shape_dtype_infer = lambda x: (x.shape[0] * 2, x.shape[1:])
  y = jax2tf.call_tf(
      lambda x: tf.concat([x, x], axis=0),
      output_shape_dype=output_shape_dtype_infer(x))(x)
```

Reply:
Actually `output_shape_dtype` cannot depend on the inputs, only on their shape. The shape of the inputs is always available to the caller and I do not see much benefit from adding that are then immediately invoked, once. If the shape computation is complex, the called can define it in a function which they call when computing the `output_shape_dtype`. This pattern has been used elsewhere in JAX, e.g., for `host_callback` and for `jax.debug` and has been sufficient for a couple of years.